drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -268,6 +268,17 @@ class Hmis::Form::Definition < ::GrdaWarehouseBase
     where(identifier: instance_scope.pluck(:definition_identifier))
   end
 
+  scope :current_versions, -> do
```

Comment:
@ttoomey, I'd like to ask for your rails expertise on this new scope, or maybe we could pair this afternoon or tomorrow. Ideally for performance purposes we would make this query using only one hit to the DB, possibly using ActiveRecord's `group` instead of ruby's `group_by`? But Gig and I weren't able to figure that out on a brief pairing session earlier today.

Another issue with the code as written is that this scope doesn't chain in an expected way with other scopes. `Hmis::Form::Definition.current_versions.non_static` returns different results from `Hmis::Form::Definition.non_static.current_versions`.

Reply:
Hi @martha 
I think you could do achieve this with postgres "DISTINCT ON" syntax (also possible with "MAX/GROUP BY" but I think distinct on is more readable). I think you could use something like the following to push this behavior into a single database query. Using a subquery is less likely to break active record's expectations than say manipulating the select clause of the query itself

```ruby
 # the id of the row with the largest version by identifier  
 CURRENT_VERSION_SQL = <<~SQL
    SELECT DISTINCT ON (identifier) id
    FROM hmis_form_definitions
    ORDER BY identifier, version DESC
  SQL
  scope :current_versions, -> {
    where("hmis_form_definitions.id IN (#{CURRENT_VERSION_SQL})")
  }
```

Regarding your second question about the difference in behavior, I think it may be because you are returning a new scope and not "chaining" from the the current scope (the current scope is `self` in this context). So you'd probably want:
```diff
-  Hmis::Form::Definition.where(id: ids)
+  self.where(id: ids)
```

Happy to pair on this if you have questions

Reply:
Wonderful @ttoomey thanks for taking a look and putting me on the right track!

Reply:
Marvelous, thank you Elliot. I think going with this helper simplified the code significantly.

(We decided against some complicated logic creating a scope that returns a published version for that identifier if it exists, else a draft if it exists, else a retired version, and instead just implemented a scope that uses `one_for_column` out-of-the-box to return the latest version per identifier.)